test/tools/Makefile

```diff
@@ -3,13 +3,23 @@
 
 .PHONY: clean provisioner-cli
 
-ARCH        ?= $(subst x86_64,amd64,$(shell uname -m))
-GOFLAGS     ?= "-tags=azure,ibmcloud,libvirt"
-BINARIES    := caa-provisioner-cli
+ARCH                    ?= $(subst x86_64,amd64,$(shell uname -m))
+BUILTIN_CLOUD_PROVIDERS ?= aws azure ibmcloud vsphere libvirt
+GOFLAGS                 ?=
+GOOPTIONS               ?= GOOS=linux GOARCH=$(ARCH) CGO_ENABLED=0
+BINARIES                := caa-provisioner-cli
```

Comment:
The binary name is `caa-provisioner-cli`, but the target name is `provisioner-cli`. Why do you use different names?

Reply:
@yoheiueda I want to put the output binary in folder `test/tools` which is same as the source code folder name, there is a known issues that the program must be launched in this folder. So I used a different binary name.

Reply:
In that case, you can write the rules like this.  Possibly, the target `provisioner-cli` is not necessary.

```
.PHONY: all clean force
BINARIES := caa-provisioner-cli

all: $(BINARIES)

.PHONY: provisioner-cli
provisioner-cli: caa-provisioner-cli

caa-provisioner-cli: force ## Build provisioner-cli for developer
	$(GOOPTIONS) go build $(GOFLAGS) -o caa-provisioner-cli "provisioner-cli/main.go"
```

Another option is to put the source code under `test/tools/cmd/provisioner-cli`.

Reply:
Thanks @yoheiueda , yes, the approach make it easier when add new binaries in test/tools, revised it.